catalyst/callbacks/batch_transform.py

```diff
@@ -1,171 +1,232 @@
-                batch_size=32,
-            ),
-            "valid": DataLoader(
-                MNIST(os.getcwd(), train=False, transform=ToTensor()),
-                batch_size=32,
-            ),
-        }
-        transforms = [
-            augmentation.RandomAffine(degrees=(-15, 20), scale=(0.75, 1.25)),
-        ]
-
-        runner = CustomRunner()
-
-        # model training
-        runner.train(
-            model=model,
-            optimizer=optimizer,
-            loaders=loaders,
-            logdir="./logs",
-            num_epochs=5,
-            verbose=True,
-            callbacks=[BatchTransformCallback(transforms, input_key=0)],
-        )
-
-    To apply augmentations only during specific loader e.g. only during
-    training :class:`catalyst.core.callbacks.control_flow.ControlFlowCallback`
-    callback can be used. For config API it can look like this:
-
-    .. code-block:: yaml
-
-        callbacks_params:
-          ...
-          train_transforms:
-            _wrapper:
-              name: ControlFlowCallback
-              loaders: train
-            name: BatchTransformCallback
-            transform:
-              - transform: kornia.RandomAffine
-                degrees: [-15, 20]
-                scale: [0.75, 1.25]
-                return_transform: true
-              - transform: kornia.ColorJitter
-                brightness: 0.1
-                contrast: 0.1
-                saturation: 0.1
-                return_transform: false
-            input_key: image
-          ...
-
-    .. _`Kornia: an Open Source Differentiable Computer Vision Library
-        for PyTorch`: https://arxiv.org/pdf/1910.02190.pdf
     """
 
     def __init__(
         self,
-        transform: Sequence[Union[dict, nn.Module]],
-        input_key: Union[str, int] = "image",
-        output_key: Optional[Union[str, int]] = None,
-    ) -> None:
-        """Init."""
-        super().__init__(order=CallbackOrder.Internal, node=CallbackNode.all)
+        lambda_fn: Callable,
+        scope: str,
+        input_key: Union[List[str], str, int] = None,
+        output_key: Union[List[str], str, int] = None,
+    ):
+        """
+        Preprocess your batch with specified function.
 
+        Args:
+            lambda_fn (Callable): Function to apply.
+            scope (str): ``"on_batch_end"`` or ``"on_batch_start"``
+            input_key (Union[List[str], str], optional): Keys in batch dict to apply function.
+            output_key (Union[List[str], str], optional): Keys for output.
+                If None then will apply function inplace to ``keys_to_apply``.
+                Defaults to ``None``.
+
+        Raises:
+            TypeError: When keys_to_apply is not str or list.
+        """
+        super().__init__(order=CallbackOrder.Internal)
+        if input_key is not None:
+            if not isinstance(input_key, (list, str, int)):
+                raise TypeError("keys to apply should be str or list of str.")
```

Comment:
int? could we add a test for this case?

Reply:
btw, not sure if we really need the `int` branch – it could be too complicated as for me to support custom tuples